src/providers/proxy/proxy_child.c

```diff
@@ -514,8 +514,8 @@ int main(int argc, const char *argv[])
         }
     }
 
-    if (domain == NULL) {
-        fprintf(stderr, "\nMissing option, "
+    if (!is_syntactically_correct_domain_name(domain)) {
+        fprintf(stderr, "\nMissing or malformed option, "
```

Comment:
Why fprintf() here instead one of our DEBUG() functions?

Reply:
This is original code. I didn't touch it since it's out of the scope of this PR.